### PR TITLE
Add benchmark notes

### PR DESCRIPTION
**Pull request by eatonphil**

### CODE REVIEW

File: src/benchmark.zig [hunk 1]
```diff
@@ -1,3 +1,24 @@
+// This benchmark uses a single client to 1) create `account_count`
```
Comment (matklad) [inline src/benchmark.zig]: Let's use `//!` to make this into a doc comment.

Comment (matklad) [inline src/benchmark.zig]: Let's maybe start with a big picture, how this file fits with the rest of the codebase (is this a micro benchmark? a macro benchmark? a whole bench or a part), and only then go into the detail of what actually this benchmarks. 

Something like 

> benchmark.zig is a program which connects to a tigerbeetle cluster and runs a predefined load against it, measuring observed latencies and throughput. See benchmark.{sh,bat} for the full benchmarking setup.

**Review by matklad — DISMISSED** [review]
Leave some comments, but that's awesome as it. 

At some point, I feel we should add a lint rule that enforces that each `.zig` file starts with `//!` which explains why it exists and what does it do.

File: src/benchmark.zig (hunk 1 above)
Comment (eatonphil) [reply src/benchmark.zig]: TIL about this. Thanks!

Comment (eatonphil) [reply src/benchmark.zig]: Good idea! Done.

Comment (eatonphil) [reply src/benchmark.zig]: Requesting a re-review after taking your suggestions.

**Review by matklad — APPROVED** [review]
:heart:

**Pull request merged**